### ISSUE TITLE
Android Support

### ISSUE THREAD

**Issue by lnanek** [LachlanMcKee/gsonpath#57]
This library looks really handy! Thanks for writing it. It would be great if it was usable with Android. Right now if I add this to my build.gradle:

```
compile('net.lachlanmckee:gsonpath:1.6.2')
```

I get this error:

```
Dex: Error converting bytecode to dex:
Cause: Dex cannot parse version 52 byte code.
This is caused by library dependencies that have been compiled using Java 8 or above.
If you are using the 'java' gradle plugin in a library submodule add
targetCompatibility = '1.7'
sourceCompatibility = '1.7'
to that submodule's build.gradle file.
```

So it either needs to be compiled with JDK 7 instead of 8, or the source target feature used to generate a compatible JAR file...

**Comment by LachlanMcKee** [comment]
Hey @lnanek,

Thanks for raising this bug. I have just encountered it myself.... For now, you should be able to use 1.6.1 without any issues. I need to investigate what I changed between the two versions to cause this incompatibility.

I am an Android developer myself and have been using 1.6.1 without any problems. Let me know if that works for you. I will have to release a new version over the weekend.

Cheers,

Lachlan

**Comment by LachlanMcKee** [comment]
So it turns out that the `net.lachlanmckee:gsonpath` artifact does not add `sourceCompatibility`, but the compiler does.

You could always temporarily use the dependencies as follows, as I have not made any changes to the core library in the latest release:

```gradle
compile 'net.lachlanmckee:gsonpath:1.6.1'
apt 'net.lachlanmckee:gsonpath-compiler:1.6.2'
```

I will push out 1.6.3 tomorrow to resolve the issue!

**Comment by lnanek** [comment]
Using 1.6.1 does seem to work. Thanks for the help!

**Comment by LachlanMcKee** [comment]
Hi @lnanek 

Sorry for the delay. I have now published version 1.6.3 which will now allow both artifacts to work when building Android apps!
